### PR TITLE
Entity Framework Dynamic LINQ support

### PR DESCRIPTION
**Pull request by ejsmith**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -51,4 +51,5 @@ _NCrunch_*
 # Rider auto-generates .iml files, and contentModel.xml
 **/.idea/**/*.iml
 **/.idea/**/contentModel.xml
-**/.idea/**/modules.xml
\ No newline at end of file
+**/.idea/**/modules.xml
```
Comment (niemyjski) [inline .gitignore:54]: we should get rid of all of this and just add `.idea`

File: build/common.props [hunk 2]
```diff
@@ -1,7 +1,7 @@
 <Project>
 
   <PropertyGroup>
-    <TargetFrameworks>netstandard2.0</TargetFrameworks>
+    <TargetFrameworks>netstandard2.0;</TargetFrameworks>
```
Comment (niemyjski) [inline build/common.props:4]: revert?

Comment (ejsmith) [reply build/common.props:4]: Nope. We are overriding TargetFrameworks and if you don't have multiple or a ; it gives a warning.

File: src/Foundatio.Parsers.SqlQueries/Extensions/SqlNodeExtensions.cs [hunk 3]
```diff
@@ -0,0 +1,297 @@
...
+
+    private static void AppendField(StringBuilder builder, EntityFieldInfo field, string term)
+    {
+        if (field == null)
+            return;
+
+        if (field.IsNumber || field.IsBoolean)
+            builder.Append(term);
+        else if (field is { IsDate: true })
+            builder.Append("DateTime.Parse(\"" + term + "\")");
```
Comment (niemyjski) [inline src/Foundatio.Parsers.SqlQueries/Extensions/SqlNodeExtensions.cs:210]: I question the name ToSqlString, when this isn't straight sql but a EF/Linq DSL.

File: src/Foundatio.Parsers.SqlQueries/SqlQueryParser.cs [hunk 4]
```diff
@@ -0,0 +1,183 @@
...
+                IsNumber = property.ClrType.UnwrapNullable().IsNumeric(),
+                IsDate = property.ClrType.UnwrapNullable().IsDateTime(),
+                IsBoolean = property.ClrType.UnwrapNullable().IsBoolean(),
+                IsCollection = isCollection
+            });
+        }
+
+        foreach (var nav in entityType.GetNavigations())
+        {
+            string propertyPath = prefix + nav.Name;
```
Comment (niemyjski) [inline src/Foundatio.Parsers.SqlQueries/SqlQueryParser.cs:129]: move closer to usage, also looks like we are missing a concatenation below.

File: src/Foundatio.Parsers.SqlQueries/SqlQueryParser.cs [hunk 5]
```diff
@@ -0,0 +1,183 @@
...
+
+            AddEntityFields(fields, skipNav.TargetEntityType, entityTypeStack, propertyPath + ".", true, depth + 1);
+        }
+
+        entityTypeStack.Pop();
+    }
+
+    private void SetupQueryVisitorContextDefaults(IQueryVisitorContext context)
+    {
+        if (!context.Data.ContainsKey("@OriginalContextResolver"))
```
Comment (niemyjski) [inline src/Foundatio.Parsers.SqlQueries/SqlQueryParser.cs:150]: Would it be worth having known consts for this?

File: tests/Directory.Build.props [hunk 6]
```diff
@@ -2,14 +2,15 @@
   <Import Project="..\build\common.props" />
   <PropertyGroup>
     <TargetFrameworks>net8.0</TargetFrameworks>
+    <TargetFrameworks>net8.0;</TargetFrameworks>
```
Comment (niemyjski) [inline tests/Directory.Build.props:5]: defined twice

File: tests/Foundatio.Parsers.SqlQueries.Tests/Foundatio.Parsers.SqlQueries.Tests.csproj [hunk 7]
```diff
@@ -0,0 +1,14 @@
...
+        <!--<DefineConstants>ENABLE_TRACING</DefineConstants>-->
+    </PropertyGroup>
+
+    <ItemGroup>
+        <ProjectReference Include="..\..\src\Foundatio.Parsers.SqlQueries\Foundatio.Parsers.SqlQueries.csproj" />
+    </ItemGroup>
+
+    <ItemGroup>
+      <PackageReference Include="Microsoft.EntityFrameworkCore.SqlServer" Version="8.0.2" />
+    </ItemGroup>
```
Comment (niemyjski) [inline tests/Foundatio.Parsers.SqlQueries.Tests/Foundatio.Parsers.SqlQueries.Tests.csproj:12]: Wouldn't this be transitive?

**Review by niemyjski — APPROVED** [review]
Looks good, left some comments for improvement

File: tests/Foundatio.Parsers.SqlQueries.Tests/Foundatio.Parsers.SqlQueries.Tests.csproj (hunk 7 above)
Comment (ejsmith) [reply tests/Foundatio.Parsers.SqlQueries.Tests/Foundatio.Parsers.SqlQueries.Tests.csproj:12]: No, project is only using abstractions

File: src/Foundatio.Parsers.SqlQueries/Extensions/SqlNodeExtensions.cs (hunk 3 above)
Comment (ejsmith) [reply src/Foundatio.Parsers.SqlQueries/Extensions/SqlNodeExtensions.cs:210]: I changed it to ToDynamicLinqString

File: src/Foundatio.Parsers.SqlQueries/SqlQueryParser.cs (hunk 5 above)
Comment (ejsmith) [reply src/Foundatio.Parsers.SqlQueries/SqlQueryParser.cs:150]: This is the same as the other code in this project. Don't want to change it now.

File: .gitignore (hunk 1 above)
Comment (ejsmith) [reply .gitignore:54]: That's not the same thing. This is the recommended gitignore from jetbrains

**Pull request merged**